great_expectations/data_context/types/base.py

```diff
@@ -436,6 +479,28 @@ def validate_schema(self, data, **kwargs):
 continue.
                 """
             )
+        if ("container" in data or "name_starts_with" in data) and not (
+            data["class_name"]
+            in [
+                "InferredAssetAzureDataConnector",
+                "ConfiguredAssetAzureDataConnector",
+            ]
+        ):
+            raise ge_exceptions.InvalidConfigError(
+                f"""Your current configuration uses one or more keys in a data connector, that are required only by an
+Azure type of the data connector (your data connector is "{data['class_name']}").  Please update your configuration to
+continue.
+                    """
+            )
+        if not ("conn_str" in data ^ "account_url" in data) and data["class_name"] in [
+            "InferredAssetAzureDataConnector",
+            "ConfiguredAssetAzureDataConnector",
+        ]:
+            raise ge_exceptions.InvalidConfigError(
+                f"""Your current configuration is either missing methods of authentication or is using too many for the Azure type of data connector.
+                You must only select one between `conn_str` and `account_url`. Please update your configuration to continue.
+                """
```

Comment:
XOR to say that we need one or the other. Is this readable?

Reply:
I really like it :) and yes it's readable